libs/client/API.lua

```diff
@@ -232,7 +233,12 @@ function API:commit(method, url, req, payload, retries)
 
 		end
 
-		client:error('%i - %s : %s %s', res.code, res.reason, method, url)
+		local errMsg = f('%i - %s : %s %s', res.code, res.reason, method, url)
+		if self._logFullErrors then
+			errMsg = errMsg .. "\n" .. msg
```

Comment:
Shouldn't need to concat here. Take advantage of the logger's formatter.

Reply:
I'm not sure what you mean. Something like
```lua
if logFullErrors then
    client:error(errMsg)
else
    client:error("%s\n%s", errMsg, msg)
end
```
?